src/homework/fifth/Calculator.java

```diff
@@ -0,0 +1,31 @@
+package homework.fifth;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Calculator {
+    private Map<Character, Calculation> operations = new HashMap<>();
+
+    public Calculator() {
+        operations.put('+', (a, b) -> a + b);
+        operations.put('-', (a, b) -> a - b);
+        operations.put('*', (a, b) -> a * b);
+        operations.put('/', (a, b) -> {
+            if (b == 0) {
+                throw new IllegalArgumentException("Division by zero");
+            }
+            return a / b;
+        });
+        operations.put('^', (a, b) -> Math.pow(a, b));
```

Comment:
optional: here you can use method referance instead of lambda. read about in which cases you can use method referance